app/src/androidTest/java/com/team41/boromi/AuthenticationControllerTest.java

```diff
@@ -0,0 +1,247 @@
+import com.team41.boromi.controllers.AuthenticationController;
+import com.team41.boromi.dbs.UserDB;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import static com.team41.boromi.constants.CommonConstants.DB_TIMEOUT;
+import static com.team41.boromi.dagger.BoromiModule.user;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@RunWith(AndroidJUnit4.class)
+public class AuthenticationControllerTest {
+
+	private final static String TAG = "AUTHENTICATION_CONTROLLER_TEST";
+
+	ArrayList<Pair<String, String>> testUsers =  new ArrayList<>();
+
+	FirebaseAuth auth = FirebaseAuth.getInstance();
+	UserDB userDB =  new UserDB(FirebaseFirestore.getInstance());
+	FirebaseFirestore db = FirebaseFirestore.getInstance();
+	ExecutorService executor = new ThreadPoolExecutor(
+			1,
+			4,
+			DB_TIMEOUT,
+			TimeUnit.MILLISECONDS,
+			new LinkedBlockingDeque<Runnable>()
+	);
+
+	AuthenticationController authController;
+
+	@Before
+	public void setup() throws InterruptedException {
+		authController = new AuthenticationController(userDB, auth, executor);
+
+		Pair<String, String> emailAndPassword = new Pair<>(
+				"testauthcontroller1@brockchelle.com",
+				"testuser"
+		);
+
+		authController.makeSignUpRequest(
+				"testauthcontroller1",
+				emailAndPassword.first,
+				emailAndPassword.second,
+				new AuthCallback() {
+					@Override
+					public void onSuccess(AuthResult authResult) {
+						Log.d(TAG, "Sign up success");
+					}
+
+					@Override
+					public void onFailure(Exception exception) {
+						Log.e(TAG, "Sign up failure");
+					}
+				}
+		);
+
+		testUsers.add(emailAndPassword);
+
+		Thread.sleep(DB_TIMEOUT);
+	}
+
+	@After
+	public void tearDown() throws InterruptedException {
+		// Loop through all the added users
+		for (Pair<String, String> emailAndPassword : testUsers) {
+			// Login first (firebase won't let you delete without logging in first)
+			authController.makeLoginRequest(
+					emailAndPassword.first,
+					emailAndPassword.second,
+					new AuthCallback() {
+						@Override
+						public void onSuccess(AuthResult authResult) {
+							Log.d(TAG, "Log in success");
+						}
+
+						@Override
+						public void onFailure(Exception exception) {
+							Log.e(TAG, "Log in failure");
+						}
+					});
+
+			Thread.sleep(DB_TIMEOUT);
+
+			// Deletes the user and their db entry
+			auth.getCurrentUser().delete()
+					.addOnCompleteListener(new OnCompleteListener<Void>() {
+						@Override
+						public void onComplete(@NonNull Task<Void> task) {
+							if (task.isSuccessful()) {
+								Log.d(TAG, "Account deleted");
+
+								try {
+									Tasks.await(

```

Comment:
not exactly sure how this works but if you delete your user first will firebase still allow you to delete from userdb?

Reply:
Yep, firebase will allow you to do that.
Technically this could fail and the user would be deleted from auth and stay in dB since it's not performed as a transaction. But since it's just for testing I tried not to get too technical.